File: 557. Reverse Words in a String III.cpp
void reverse(int left, int right, char *s){
    while(left < right) {
        s[left]^=s[right];
        s[right]^=s[left];
        s[left]^=s[right];
        left++;
        right--;
    }
}

char* reverseWords(char* s) {
    int i,len=strlen(s),index = 0;
    
    for(i=0;i<=len;i++) {
        if((s[i] == ' ') || (s[i] == '\0')){
            reverse(index, i - 1, s);
            index = i + 1;
        }
    }
    return s;
}

File: Type/LeetCode_String/Easy/557. Reverse Words in a String III.java
class Solution {
    public String reverseWords(String s) 
{
    char[] c = s.toCharArray();
    int i = 0;
    for(int j = 0; j < c.length; j++)
    {
        if(c[j] == ' ')
        {
            reverse(c, i, j - 1);
            i = j + 1;
        }
    }
    reverse(c, i, c.length - 1);
    return new String(c);
}

public void reverse(char[] s, int l, int r)
{
    while(l<r)
    {
        s[l]^=s[r];
        s[r]^=s[l];
        s[l]^=s[r];
        l++;r--;
    }
}
}